### PR TITLE
Feature/page embedding

### PR DESCRIPTION
**Pull request by wotwot563**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [ ] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Do something here...
2. Then run this test...
3. After that, you should see the result here

## Link to issue

Closes: #56

### CODE REVIEW

**Comment by wotwot563** [comment]
as requested by brend:
For the frontend there needs to be a button that generates the embeddedproject and shows the iframe code:

here is an example 
<iframe src="https://dex.software/embed/dd892529-814c-41ff-ad73-e6f8f8402300" style="border:0px #ffffff none;" name="myiFrame" scrolling="no" frameborder="1" marginheight="0px" marginwidth="0px" height="400px" width="600px" allowfullscreen></iframe>

the guid is custom per project and the embed endpoint should be a public non auth frontend endpoint so that everyone can access it

**Comment by StijnGroenen** [comment]
Can you add a description to this PR describing what it adds?
Can't the front end just create a new embed page component (For example https://dex.software/embed/{PROJECT_ID} ) using the data from the existing project endpoints (since they are all public endpoints)?
What's the benefit of the code you added to the back end?

**Comment by wotwot563** [comment]
The main benefit this has is that when for example fontys wants to showcase a project it can do so without authorization, with this safeguard only approved users or the creator of the project can create an embedded project.

**Comment by StijnGroenen** [comment]
Okay.
I thought that since the projects are all publically visible, authorization wouldn't be a problem in the first place. But if you want to restrict who can create embeds than it makes sense.

File: API/Controllers/EmbedController.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        private readonly IEmbedService embedService;
+        private readonly IMapper mapper;
+        private readonly IProjectService projectService;
+        private readonly IUserService userService;
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmbedController"/> class.
+        /// </summary>
+        /// <param name="embedService">The embed service.</param>
+        /// <param name="mapper">The mapper.</param>
+        /// <param name="projectService"></param>
```
Comment (StijnGroenen) [inline API/Controllers/EmbedController.cs:52]: Empty param tags should be filled in.

File: API/Controllers/EmbedController.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                };
+                return NotFound(problem);
+            }
+            return Ok(mapper.Map<Project, ProjectResourceResult>(embeddedProject.Project));
+        }
+
+        /// <summary>
+        ///     Creates a embedded project
+        /// </summary>
+        /// <param name="embedResource"></param>
```
Comment (StijnGroenen) [inline API/Controllers/EmbedController.cs:113]: Empty param tags should be filled in.

File: Data/ApplicationDbContext.cs [hunk 3]
```diff
@@ -37,6 +37,8 @@ public class ApplicationDbContext : DbContext
 
         public DbSet<Highlight> Highlight { get; set; }
 
+        public DbSet<EmbeddedProject> EmbeddedProject { get; set; }
```
Comment (StijnGroenen) [inline Data/ApplicationDbContext.cs:40]: Missing documentation comment.

File: Repositories/EmbedRepository.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        Task<IEnumerable<EmbeddedProject>> GetEmbeddedProjectsAsync();
+        Task<bool> IsNonExistingGuid(Guid guid);
+    }
+
+    public class EmbedRepository : Repository<EmbeddedProject>, IEmbedRepository
+    {
+
+        public EmbedRepository(DbContext dbContext) : base(dbContext) { }
+
+        public async Task<EmbeddedProject> GetEmbeddedProjectAsync(Guid guid)
```
Comment (StijnGroenen) [inline Repositories/EmbedRepository.cs:42]: Missing documentation comments.

File: Services/Services/EmbedService.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        Task<bool> IsNonExistingGuid(Guid guid);
+    }
+
+    public class EmbedService : Service<EmbeddedProject>, IEmbedService
+    {
+        public EmbedService(IEmbedRepository repository) : base(repository) { }
+
+        protected new IEmbedRepository Repository => (IEmbedRepository) base.Repository;
+
+        public async Task<EmbeddedProject> FindAsync(Guid guid)
```
Comment (StijnGroenen) [inline Services/Services/EmbedService.cs:41]: Missing documentation comments.

File: Repositories/EmbedRepository.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        /// Gets the embedded projects asynchronous.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<EmbeddedProject>> GetEmbeddedProjectsAsync();
+        /// <summary>
+        /// Determines whether [is non existing unique identifier] [the specified unique identifier].
+        /// </summary>
+        /// <param name="guid">The unique identifier.</param>
+        /// <returns></returns>
+        Task<bool> IsNonExistingGuid(Guid guid);
```
Comment (Brend-Smits) [inline Repositories/EmbedRepository.cs:47]:
```suggestion
        Task<bool> IsNonExistingGuidAsync(Guid guid);
```

File: Repositories/EmbedRepository.cs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        /// <summary>
+        /// Determines whether [is non existing unique identifier] [the specified unique identifier].
+        /// </summary>
+        /// <param name="guid">The unique identifier.</param>
+        /// <returns>
+        ///   <c>true</c> if [is non existing unique identifier] [the specified unique identifier]; otherwise, <c>false</c>.
+        /// </returns>
+        public async Task<bool> IsNonExistingGuid(Guid guid)
```
Comment (Brend-Smits) [inline Repositories/EmbedRepository.cs:97]:
```suggestion
        public async Task<bool> IsNonExistingGuidAsync(Guid guid)
```

File: Services/Services/EmbedService.cs [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        /// Gets the embedded projects asynchronous.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<EmbeddedProject>> GetEmbeddedProjectsAsync();
+        /// <summary>
+        /// Determines whether [is non existing unique identifier] [the specified unique identifier].
+        /// </summary>
+        /// <param name="guid">The unique identifier.</param>
+        /// <returns></returns>
+        Task<bool> IsNonExistingGuid(Guid guid);
```
Comment (Brend-Smits) [inline Services/Services/EmbedService.cs:46]:
```suggestion
        Task<bool> IsNonExistingGuidAsync(Guid guid);
```

File: Services/Services/EmbedService.cs [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        public async Task<IEnumerable<EmbeddedProject>> GetEmbeddedProjectsAsync()
+        {
+            return await Repository.GetEmbeddedProjectsAsync();
+        }
+        /// <summary>
+        /// Determines whether [is non existing unique identifier] [the specified unique identifier].
+        /// </summary>
+        /// <param name="guid">The unique identifier.</param>
+        /// <returns></returns>
+        public async Task<bool> IsNonExistingGuid(Guid guid)
```
Comment (Brend-Smits) [inline Services/Services/EmbedService.cs:89]:
```suggestion
        public async Task<bool> IsNonExistingGuidAsync(Guid guid)
```

File: Services/Services/EmbedService.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            return await Repository.IsNonExistingGuid(guid);
```
Comment (Brend-Smits) [inline Services/Services/EmbedService.cs:91]:
```suggestion
            return await Repository.IsNonExistingGuidAsync(guid);
```

**Comment by Brend-Smits** [comment]
Basic unit testing framework is now set up by @jcoenen96 , this makes me believe that unit tests can be added. Can you look into it? Best if we implement it right away instead of backtracking.

**Pull request merged**